src/main/java/com/amazon/opendistroforelasticsearch/sql/antlr/semantic/visitor/ESMappingLoader.java

```diff
@@ -105,9 +105,25 @@ private void defineIndexType(String indexName) {
 
     private void loadAllFieldsWithType(String indexName) {
         FieldMappings mappings = getFieldMappings(indexName);
+        saveDateFormats(mappings);
         mappings.flat(this::defineFieldName);
     }
 
+    private void saveDateFormats(FieldMappings mappings) {
```

Comment:
I think you may not need to touch this class or the `LocalClusterState` either. Because `LocalClusterState` has the mapping, you can just use it outside or in your new added Formatter class.

Reply:
Now fetching information from `LocalClusterState` in my Formatter class

Reply:
Any chance to avoid the changes in `LocalClusterState`? It is a global state which can not be used for state per query.

Reply:
Moved changes out of `LocalClusterState`